frontend/app/src/components/helper/BlockchainAccountSelector.vue

```diff
@@ -0,0 +1,201 @@
+import { createNamespacedHelpers } from 'vuex';
+import CryptoIcon from '@/components/CryptoIcon.vue';
+import TagIcon from '@/components/tags/TagIcon.vue';
+
+import {
+  Account,
+  AccountData,
+  AccountDataMap,
+  Tags,
+  Blockchain
+} from '@/typing/types';
+
+export interface GeneralAccount extends AccountData {
+  chain: Blockchain;
+}
+
+const { mapState } = createNamespacedHelpers('session');
+const { mapState: mapBalanceState } = createNamespacedHelpers('balances');
+
+@Component({
+  components: { CryptoIcon, TagIcon },
+  computed: {
+    ...mapState(['tags']),
+    ...mapBalanceState(['ethAccounts', 'btcAccounts'])
+  }
+})
+export default class BlockchainAccountSelector extends Vue {
+  @Prop({ required: true, type: Array, default: [] })
+  addresses!: Account[];
+  @Prop({ required: false, type: Boolean, default: false })
+  multiple!: boolean;
+
+  ethAccounts!: AccountDataMap;
+  btcAccounts!: AccountDataMap;
+  tags!: Tags;
+  selectedAccounts: GeneralAccount[] | GeneralAccount | null = null;
+  selectedAccountsArray: GeneralAccount[] = [];
+  select = ['Vuetify', 'Programming'];
+  items = ['Programming', 'Design', 'Vue', 'Vuetify'];
+  search: string = '';
+
+  get filteredBlockchainAccounts(): any[] {
+    let filteredAccounts: GeneralAccount[] = [];
+
+    // filter the addresses for each blockchain then group them into filteredAddresses
+    for (let [key, value] of Object.entries(this.ethAccounts)) {
+      if (this.addresses[this.addresses.findIndex(x => x.address === key)]) {
+        filteredAccounts.push({ chain: 'ETH', ...value });
+      }
+    }
+    for (let [key, value] of Object.entries(this.btcAccounts)) {
+      if (this.addresses[this.addresses.findIndex(x => x.address === key)]) {
+        filteredAccounts.push({ chain: 'BTC', ...value });
+      }
+    }
+
+    return filteredAccounts;
+  }
+
+  @Watch('selectedAccounts')
+  onSelectedAccountsChange() {
+    // When the component isn't in 'multiple' mode we have to make sure that we're still sending
+    // either an empty array or an array of one back so that the filterable methods do not break.
+    if (
+      this.selectedAccounts &&
+      typeof this.selectedAccounts === 'object' &&
+      !Array.isArray(this.selectedAccounts)
+    ) {
+      this.selectedAccountsArray = [];
+      this.selectedAccountsArray.push(this.selectedAccounts as GeneralAccount);
+    } else if (!this.selectedAccounts) {
+      this.selectedAccountsArray = [];
+    } else {
+      this.selectedAccountsArray = this.selectedAccounts;
+    }
+    this.$emit('selected-accounts-change', this.selectedAccountsArray);
+
+    // Force clear the search value on change (e.g. after searching for an account
+    // and then clicking enter)
+    if (this.search) {
+      this.search = '';
+    }
+  }
+
+  filter(item: GeneralAccount, queryText: string) {
+    const hasValue = (val: string | null) => (val != null ? val : '');
```

Comment:
do we really need this has value here?

Reply:
This was really time consuming to get right so once it worked I didn't dare touch it . I followed the vuetify custom filter example which used hasValue and whatever you seen in the below 2 comments you made.

Reply:
labels are always either empty string or string so no need to check them